docker/files/run/run.sh

```diff
@@ -95,20 +116,24 @@ function start_new_public_network() {
 
   local ID
   case $NETWORK in
-    mainnet)  ID="<network>.algorand.network";;
-    testnet)  ID="<network>.algorand.network";;
-    betanet)  ID="<network>.algodev.network";;
-    alphanet) ID="<network>.algodev.network";;
-    devnet)   ID="<network>.algodev.network";;
-    *)        echo "Unknown network"; exit 1;;
+  mainnet) ID="<network>.algorand.network" ;;
+  testnet) ID="<network>.algorand.network" ;;
+  betanet) ID="<network>.algodev.network" ;;
+  alphanet) ID="<network>.algodev.network" ;;
+  devnet) ID="<network>.algodev.network" ;;
+  *)
+    echo "Unknown network"
+    exit 1
+    ;;
   esac
   set -p DNSBootstrapID -v "$ID"
 
   start_public_network
 }
 
 function start_private_network() {
-  apply_configuration
+  configure_data_dir
+  start_kmd &
```

Comment:
No need to run in background if you start kmd using goal
```suggestion
  start_kmd
```